Add SmartPress to contract create options (#56)
0xPolygon/wiki: docs/pos/deploy/smartpress.md

---
id: smartpress
title: How to Deploy a Smart Contract Using SmartPress
sidebar_label: How to Deploy a Smart Contract Using SmartPress

> description: "Lean how to deploy smart contracts on Polygon PoS mainnet using SmartPress."

Reviewer: ```suggestion
description: "Learn how to deploy smart contracts on Polygon PoS mainnet using SmartPress."
```

keywords:
  - docs
  - matic
  - polygon
  - smartpress
  - ai
  - create smart contract
  - deploy on polygon
image: https://wiki.polygon.technology/img/polygon-logo.png
---

:::caution Content disclaimer

Please view the third-party content disclaimer [<ins>here</ins>](https://github.com/0xPolygon/wiki/blob/master/CONTENT_DISCLAIMER.md).

:::

> SmartPress is a smart contract development ai tool that builds your dream contract from a language prompt.

Reviewer: ```suggestion
SmartPress is an AI tool that crafts custom smart contracts from language prompts.
```

> > For more information on different contracts SmartPress can build, check out [smartpress’s website](https://smartpress.ai).

Reviewer: ```suggestion
> For details on the variety of contracts SmartPress can create, check out [SmartPress’s website](https://smartpress.ai).
```

> ## Create a contract using SmartPress

Reviewer: ```suggestion
## Create a Smart Contract using SmartPress
```

To create a new smart contract using SmartPress, follow the steps below.

1. In your browser, navigate to smartpress.ai

> 2. Enter the requirements for the contract you want to build.

Reviewer: ```suggestion
2. Enter the specifications for your desired contract.
```

> 3. SmartPress will handle it from there and produce your contract for you without you having to use a IDE or a code editor.

Reviewer: ```suggestion
3. Let SmartPress generate the contract for you; no need for an IDE or code editor.
```

4. If you are happy with the output, move onto validating your idea and testing. If you arent happy, edit your prompt and start again.

![Screenshot 2023-08-15 at 12 33 59 pm](https://github.com/crokau/wiki/assets/71380821/c43ccb48-3b1b-4cad-814b-8eddc0f735c1)

> ## Deploy your contract using SmartPress

Reviewer: ```suggestion
## Deploy your Contract using SmartPress

To deploy your smart contract on Polygon PoS:
```

> 1. Once created, connect your wallet, change your network to Polygon Testnet/Mainnet.

Reviewer: ```suggestion
1. After creation, connect your wallet and switch your network to either Mumbai testnet or Polygon PoS mainnet.
```

2. Hit the deploy button, or use the faucet links to get transaction gas if you dont have any yet.

![Screenshot 2023-08-15 at 12 37 39 pm](https://github.com/crokau/wiki/assets/71380821/ea20889b-1b5e-44b7-ba9c-f208abf1e944)

## Verify the Contract

> Press 'view code', copy flattened bytecode to clipboard, in your preffered block explorer navigate to verification screen (usually under contracts/code), select compiler version specified in the deployment page, optimizer: TRUE, Runs: 200, choose preffered licence, paste flat code, verification should complete shortly.

Reviewer: I believe this section can also be presented as steps:

1. Click on 'View Code' and copy the flattened bytecode to your clipboard.
2. In your preferred block explorer, navigate to the verification screen (typically found under 'contracts/code').
3. Select the compiler version as indicated on the deployment page.
4. Set the optimizer to 'TRUE' and runs to '200'.

and so on.

Let's also add a starting statement: "To verify your new contract..."

keywords:
  - docs
  - matic
  - polygon
  - smartpress
  - ai
  - create smart contract
  - deploy on polygon
image: https://wiki.polygon.technology/img/polygon-logo.png
---

:::caution Content disclaimer

Please view the third-party content disclaimer [<ins>here</ins>](https://github.com/0xPolygon/wiki/blob/master/CONTENT_DISCLAIMER.md).

:::

SmartPress is an AI tool that crafts custom smart contracts from language prompts.

> For details on the variety of contracts SmartPress can create, check out [SmartPress’s website](https://smartpress.ai).

## Create a Smart Contract using SmartPress

To create a new smart contract with SmartPress, follow these steps:

1. In your browser, navigate to [smartpress.ai](smartpress.ai).
2. Enter the specifications for your desired contract.
3. Let SmartPress generate the contract for you; no need for an IDE or code editor.
4. Review the output. If satisfied, proceed to validation and testing. If not, modify your input and repeat the process.

![Screenshot 2023-08-15 at 12 33 59 pm](https://github.com/crokau/wiki/assets/71380821/c43ccb48-3b1b-4cad-814b-8eddc0f735c1)

## Deploy your Contract using SmartPress

> To deploy your smart contract on Polygon PoS:

Reviewer: ```suggestion
To deploy your smart contract on Polygon PoS:

```

1. After creation, connect your wallet and switch your network to either Mumbai testnet or Polygon PoS mainnet.
2. Press the 'Deploy' button. If you lack transaction gas, use the provided faucet links to obtain some.

![Screenshot 2023-08-15 at 12 37 39 pm](https://github.com/crokau/wiki/assets/71380821/ea20889b-1b5e-44b7-ba9c-f208abf1e944)

## Verify the Contract

> 1. To verify your new contract, start by clicking on 'View Code' and copy the flattened version of your code to the clipboard.

Reviewer: ```suggestion

To verify your new contract, follow these steps:

1. Start by clicking on 'View Code' and copy the flattened version of your code to the clipboard.
```
